extensions/statedb/store.go

```diff
@@ -0,0 +1,16 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package privacyenabledstate
+
+import (
+	"github.com/hyperledger/fabric/core/ledger/kvledger/txmgmt/statedb"
+)
+
+// NewVersionedDBProvider instantiates VersionedDBProvider
+func NewVersionedDBProvider(vdbProvider statedb.VersionedDBProvider) statedb.VersionedDBProvider {
```

Comment:
It's kind of weird to call NewVersionedDBProvider while returning the provider that's passed in. I think it would make more sense to have NewVersionedDBProvider return either the couch provider or the level provider, i.e. move the code from common_storage_db (if ledgerconfig.IsCouchDBEnabled...) here.

Reply:
The idea here is to have independent db cache implementation irrespective of DB Provider (level/couch). In future, if fabric supports more DB Providers, we wont need to update this extension point. The real ext implementation will have cache provider bundled in along with vdbProvider passed to this function.

Reply:
Then maybe it's a matter of proper naming for this function. What you're adding is a decorator/interceptor chain?

Reply:
Agreed on name change. Better name would make more sense as this is not really creating a new VersionedDB Provider.

Reply:
I can't think of a good name. If we think of something better then we'll change it later.